src/controllers/shared.controllers/director.correspondence.link.controller.ts

```diff
@@ -0,0 +1,89 @@
+import { NextFunction, Request, Response } from "express";
+import { DIRECTOR_CORRESPONDENCE_ADDRESS_PATH, DIRECTOR_RESIDENTIAL_ADDRESS_PATH } from "../../types/page.urls";
+import { Templates } from "../../types/template.paths";
+import { urlUtils } from "../../utils/url";
+import { createValidationErrorBasic, formatValidationErrors } from '../../validation/validation';
+import { OfficerFiling } from "@companieshouse/api-sdk-node/dist/services/officer-filing";
+import { getOfficerFiling, patchOfficerFiling } from "../../services/officer.filing.service";
+import { formatTitleCase } from "../../services/confirm.company.service";
+import { retrieveDirectorNameFromFiling } from "../../utils/format";
+import { DirectorField } from "../../model/director.model";
+import { getField } from "../../utils/web";
+import { Session } from "@companieshouse/node-session-handler";
+import { SA_TO_ROA_ERROR } from "../../utils/constants";
+
+export const getCorrespondenceLink = async (req: Request, res: Response, next: NextFunction, templateName: string, backUrlPath: string,) => {
+  try {
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const session: Session = req.session as Session;
+
+    const officerFiling = await getOfficerFiling(session, transactionId, submissionId);
+
+    return res.render(templateName,{
+      templateName: templateName,
+      backLinkUrl: urlUtils.getUrlToPath(backUrlPath, req),
+      directorName: formatTitleCase(retrieveDirectorNameFromFiling(officerFiling)),
+      sa_to_roa: calculateSaToRoaRadioFromFiling(officerFiling.isServiceAddressSameAsRegisteredOfficeAddress),
+    });
+  } catch (e) {
+    return next(e);
+  }
+};
+
+export const postCorrespondenceLink = async (req: Request, res: Response, next: NextFunction, templateName: String, nextPageUrl: string, backUrlPath: string) => {
+  try {
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const session: Session = req.session as Session;
+    const isServiceAddressSameAsRegisteredOfficeAddress = calculateSaToRoaBooleanValue(req);
+
+    if (isServiceAddressSameAsRegisteredOfficeAddress === undefined) {
+      const officerFiling = await getOfficerFiling(session, transactionId, submissionId);
+      const linkError = createValidationErrorBasic(SA_TO_ROA_ERROR, DirectorField.SA_TO_ROA_RADIO);
+      return res.render(Templates.DIRECTOR_CORRESPONDENCE_ADDRESS_LINK, {
```

Comment:
This will bounce the user back to the DIRECTOR_CORRESPONDENCE_ADDRESS_LINK page always. Add a parameter to indicate what is the right TEMPLATE for the calling controller.

Reply:
Thanks, I'd missed that. Have updated now